zerver/lib/server_initialization.py

```diff
@@ -23,20 +24,21 @@ def server_initialized() -> bool:
 def create_internal_realm() -> None:
     from zerver.actions.users import do_change_can_forge_sender
 
-    realm = Realm.objects.create(string_id=settings.SYSTEM_BOT_REALM, name="System bot realm")
-    RealmAuditLog.objects.create(
-        realm=realm, event_type=RealmAuditLog.REALM_CREATED, event_time=realm.date_created
-    )
-    RealmUserDefault.objects.create(realm=realm)
-    create_system_user_groups_for_realm(realm)
+    with transaction.atomic(durable=True):
```

Comment:
What's the thinking behind doing this here rather than around all of `create_internal_realm`? The latter certainly seems more correct to me, since then either we have a fully initialized server or a server that's not initialized at all.